### PR TITLE
Upgrade vanilla 2.5.0

### PR DESCRIPTION
**Pull request by huwshimi**
## Done
- Upgraded Vanilla to 2.5.0.
- Upgrade react-components to 0.3.0.
- Replaced overrides with utilities.
- Fixed the loading position. Fixes: https://github.com/canonical-web-and-design/maas-ui/issues/524

## QA
- Load the react ui, check that the loading spinners are aligned with the header logo.
- View the settings and preferences and check that the icon buttons are unchanged in the lists (e.g. users, api keys).
- Check that all the settings pages now load.

### CODE REVIEW

File: proxy/index.js [hunk 1]
```diff
@@ -42,6 +42,9 @@ app.use(
   })
 );
 
+// Proxy the HMR url to the React client.
+app.use(proxy("/main.*.hot-update.js", { target: "http://localhost:8401/" }));
```
Comment (huwshimi) [inline proxy/index.js]: Drive-by to load the HMR js.

File: ui/src/app/settings/components/Routes/Routes.js [hunk 2]
```diff
@@ -68,16 +66,8 @@ const Routes = () => {
         to={`${match.path}/configuration/general`}
       />
       <Route exact path={`${match.path}/users`} component={UsersList} />
-      {!externalAuthURL && (
```
Comment (huwshimi) [inline ui/src/app/settings/components/Routes/Routes.js]: Drive-by to fix the URLs (this check is now done in the components).

File: proxy/index.js (hunk 1 above)
Comment (squidsoup) [reply proxy/index.js]: Nice
